sky/skylet/constants.py

```diff
@@ -156,9 +156,21 @@
     # Caller should replace {conda_auto_activate} with either true or false.
     'conda config --set auto_activate_base {conda_auto_activate} && '
     'conda activate base; }; '
+    # If conda was not installed and the image is a docker image,
+    # we deactivate any active conda environment we set.
+    # Caller should replace {is_custom_docker} with either true or false.
+    'if [ "{is_custom_docker}" = "true" ]; then '
+    'conda deactivate;'
```

Comment:
Do we even use this conda env? The skypilot venv will use it's own python from uv, from the `uv venv --python` flag.
We should really document all this behavior. I personally feel that we should also avoid setting up conda for custom AMIs.